flink-java/src/main/java/org/apache/flink/api/java/aggregation/SumAggregationFunction.java

```diff
@@ -113,11 +182,32 @@ public Long getAggregate() {
 			return agg;
 		}
 	}
-	
+
+	public static final class LongValueSumAgg extends SumAggregationFunction<LongValue> {
+		private static final long serialVersionUID = 1L;
+
+		private long agg;
+
+		@Override
+		public void initializeAggregate() {
+			agg = 0L;
+		}
+
+		@Override
+		public void aggregate(LongValue value) {
+			agg += value.getValue();
+		}
+
+		@Override
+		public LongValue getAggregate() {
+			return new LongValue(agg);
+		}
+	}
+
 	public static final class FloatSumAgg extends SumAggregationFunction<Float> {
 		private static final long serialVersionUID = 1L;
-		
-		private float agg;
+
+		private double agg;
```

Comment:
The extra precision is beneficial when locally accumulating millions to billions of value considering the 23 bits of precision. Is there a performance advantage to using float over double and removing the implicit cast?